autotest/ogr/ogr_sosi.py

```diff
@@ -55,4 +55,17 @@ def test_ogr_sosi_1():
     assert ret.find('INFO') != -1 and ret.find('ERROR') == -1
 
 
+# test that -oo appendFieldsMap 
+def test_ogr_sosi_2():
+
+    if ogr.GetDriverByName('SOSI') is None:
+        pytest.skip()
+
+    import test_cli_utilities
+    if test_cli_utilities.get_test_ogrsf_path() is None:
+        pytest.skip()
+
+    ret = gdaltest.runexternal(test_cli_utilities.get_test_ogrsf_path() + ' -ro data/test_duplicate_fields.sos -oo appendFieldsMap="BEITEBRUKERID&OPPHAV"')
```

Comment:
that's a bit overkill to run test_ogrsf which tests the general driver behaviour, but doesn't verify the working of the open option. You should likely open the dataset with the option, iterate on a feature and check the value of the field with appendFieldsMap
Relevant except from other tests
```
    ds = gdal.OpenEx('data/mvt/datatypes/0/0/0.pbf',
                     open_options=['METADATA_FILE='])
    lyr = ds.GetLayer(0)
   f = lyr.GetNextFeature()
   asser f['foo'] = 'bar'
```

Reply:
Thanks, fixed now. Not very used to Python either.